crates/release_plz_core/src/command/update/updater.rs

```diff
@@ -82,6 +82,11 @@ impl Updater<'_> {
 
         let mut old_changelogs = OldChangelogs::new();
         for (p, diff) in packages_diffs {
+            if diff.is_version_published {
+                // We need to check if one of the dependencies of this package changed.
+                packages_to_check_for_deps.push(p);
+            }
```

Comment:
I think moving this block of code here is wrong. We just need to check if a package needs to be updated because of its dependencies if `next_version != current_version`.
Otherwise we are going to update it anyway.

Imo the real issue is that the function `dependent_packages_update` doesn't detect that `binary` needs to be updated when running `TEST_LOGS=1 ENABLE_LOGS=1 RELEASE_PLZ_LOG=info cargo t release_plz_updates_binary_with_no_commits_and_dependency_change -- --nocapture`.

I spent two hours on this tonight and I'm too tired now to continue now.